src/mame/machine/nb1412m2.cpp

```diff
@@ -179,7 +178,8 @@ void nb1412m2_device::device_reset()
 	m_rom_op = 0;
 	m_const90 = 0x18; // fixes coin sample if inserted at first title screen
 	m_dac_current_address = m_dac_start_address = 0;
-	m_dac_frequency = 4000;
+	m_timer_rate = 2;
+	m_dac_frequency = m_timer_rate * 1296;
```

Comment:
Where do you believe this base DAC frequency of 1296 comes from?  What are the clock inputs to the device?  Is it more likely to be using dividers, or does it look like it has a PLL setup?

Reply:
This was guesswork purely from the pitch of the sound, and the best prime factor was chosen.
I'm sorry, but I don't have any better means.

Reply:
But I did not write a comment on this, so I added it.

Reply:
There's an 8mhz clock on the sound PCB so I suppose that's the clk input. Let's guess and divide that by 512 and you get 15625Hz(incidentally, it could be hsync clock too then?)
15625 / 35 = ~446Hz
15625 / 12 = ~1302Hz

Reply:
Thank you! I appreciate the support as I am lacking in hardware knowledge.
From this advice, I will changed frequencys to 446 & 1302.
Does this look OK?

Reply:
It already has a clock configured in the drivers
cop01.cpp: NB1412M2(config, m_prot, XTAL(8'000'000)/2); // divided by 2 maybe
galivan.cpp: NB1412M2(config, m_prot, XTAL(8'000'000)); // divided by 2 maybe
terracre.cpp: NB1412M2(config, m_prot, XTAL(16'000'000)); // divided by 4 maybe

I'd make that consistent to make them all end up as 4MHz, change comment that divider is unsure.

Then change your 448 and 1296 values to:
(clock()) / (256 * 35)
(clock()) / (256 * 12)

In device_reset(), instead of repeating yourself, do a dac_timer_w(0x10);
And by the way "Argo" is not a word, change to "Algorithm"

Reply:
I'll go ahead and change the drivers so you won't have to bother adding new files to the PR

Reply:
I made the changes as you suggested.
Thanks for the advice!

Reply:
Ok thanks